### DISCUSSION TITLE
For correlation plot

### DISCUSSION THREAD

**Discussion by Sam131992** [ChiLiubio/microeco#224] [General]
Dear Microeco team<
I have two queries regarding environmental correlation analysis:

If we assume we have 30 samples - 15 from Group A and 15 from Group B - and collected 5 soil samples at each field for physicochemical analysis, but for microbiome analysis, we combined these 5 samples into 1 composite sample. So now we have 30 physicochemical observations and 6 microbiome observations. Is it appropriate to analyze the data this way, considering that we have only 6 microbiome observations out of 30 total observations?

How many parameters can we include in the environmental correlation analysis? I have selected 20, but the plot is only showing 8 parameter observations. Can we increase the number of parameters?

**Comment by ChiLiubio** [comment]
Maybe you can get an average env value to make the samples consistent with the microbiome data. So it depends on your goal.
I am confused about that without your detailed data or figure. I guess it probably comes from the non-numeric data in your env data. Please check your data again. Make sure all required columns are numeric.

**Reply by Sam131992** [reply]
![image](https://user-images.githubusercontent.com/116187581/232316700-cbb26a38-27ad-45cd-a462-a6f52db6d764.png)
please check this image I have total 20 parameters such as pk, ec, bd, soc, som, zn, mn, cu, fe, s, tn, tp, tk, aldp, glu, DHA, aklp, urea but the image showing only 8 parameters.... how many maximum numbers we can select for this plot?

**Comment by ChiLiubio** [comment]
Though I am not sure wether GGally package has a strict on the number of variables, I have tried the default example data with 11 vars, and all things go well.
```
library(microeco)
data(dataset)
data(env_data_16S)
t1 <- trans_env$new(dataset = dataset, add_data = env_data_16S)
p <- t1$cal_autocor()
p
```
